presto-main/src/main/java/io/prestosql/metadata/FunctionResolver.java

```diff
@@ -140,13 +144,10 @@ FunctionBinding resolveFunction(Collection<FunctionMetadata> allCandidates, Qual
                     Joiner.on(", ").join(function.getSignature().getArgumentTypes()),
                     Joiner.on(", ").join(function.getSignature().getTypeVariableConstraints())));
         }
-        String parameters = Joiner.on(", ").join(parameterTypes);
-        String message = format("Function '%s' not registered", name);
-        if (!expectedParameters.isEmpty()) {
-            String expected = Joiner.on(", ").join(expectedParameters);
-            message = format("Unexpected parameters (%s) for function %s. Expected: %s", parameters, name, expected);
-        }
 
+        String parameters = Joiner.on(", ").join(parameterTypes);
+        String expected = Joiner.on(", ").join(expectedParameters);
+        String message = format("Unexpected parameters (%s) for function %s. Expected: %s", parameters, name, expected);
```

Comment:
Maybe wrap expected string into brackets too in case list is empty

Reply:
which list?  i don't think any can be empty at this point

Reply:
Why not? Previous code assumed `expectedParameters` could be empty

Reply:
Only when there `allCandidates` are empty, which is now filtered out at the top of the method.